ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala

```diff
@@ -246,6 +251,78 @@ private[appendonlydao] final class TransactionsReader(
       .map(EventsTable.Entry.toGetTransactionResponse)
   }
 
+  override def getTransactionLogUpdates(
+      startExclusive: (Offset, Long),
+      endInclusive: (Offset, Long),
+  )(implicit
+      loggingContext: LoggingContext
+  ): Source[((Offset, Long), TransactionLogUpdate), NotUsed] = {
+    val endMarker = Source.single(
+      endInclusive -> TransactionLogUpdate.LedgerEndMarker(
+        eventOffset = endInclusive._1,
+        eventSequentialId = endInclusive._2,
+      )
+    )
+
+    val eventsSource = Source
+      .fromIterator(() =>
+        TransactionsReader
+          .splitRange(
```

Comment:
when split range throws an exception we do no recover

Reply:
When it throws, it's actually unrecoverable, not much we can do about it. The error condition is actually reachable only via a programming error, so far-fetched and unrecoverable

Reply:
Furthermore, the consumers can also retry, which we actually do for `getContractStateEvents` and we will do with the consumer of this as well. Actually the good point here is to somehow take down the entire process since it can never recover. What do you think?